### PR TITLE
Presidio Versioning

### PR DESCRIPTION
**Pull request by balteravishay**
This PR adds versioning semantics to presidio:
* VERSION file is semver, ready to be merged to master (without .alpha notations).
* each PR is validated that it contains a change to the VERSION file.
* an initial release pipeline (currently manually triggered) is reading the version file to create a GitHub release.
* updated contribution guidelines and template

### CODE REVIEW

File: .pipelines/templates/validate-version.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          GIT_DIFF_UPSTREAMBRANCH=remotes/origin/$GIT_DIFF_UPSTREAMBRANCH
+        fi
+
+        GIT_DIFF_SOURCEBRANCH="HEAD"
+
+        VERSION_FILECHANGE_SET=$(git diff "$GIT_DIFF_SOURCEBRANCH" "$GIT_DIFF_UPSTREAMBRANCH" --name-only | grep -w VERSION)
+        # check if file was changeds
+        if [ -z "$VERSION_FILECHANGE_SET" ]; then
+            echo "version file not changed"
+        fi
```
Comment (SharonHart) [inline .pipelines/templates/validate-version.yml:26]: should we add format validation with regex?

File: .pipelines/templates/release.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        echo $ver
+        echo "##vso[task.setvariable variable=version;isOutput=true]$ver"
+      displayName: Set Version
+      name: setVer
+- job: githubRelease
+  displayName: GitHub Release
+  dependsOn: version
+  variables:
+    version: $[ dependencies.Version.outputs['setVer.version'] ]
+  steps:
```
Comment (SharonHart) [inline .pipelines/templates/release.yml:19]: nice! I was thinking plain build failure but this is much better

**Review by SharonHart — APPROVED** [review]

**Review by shiranr — APPROVED** [review]

File: .pipelines/templates/validate-version.yml (hunk 1 above)
Comment (navalev) [reply .pipelines/templates/validate-version.yml:26]: maybe check that the version change does not change the major? (or even minor)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ To get started, refer to the documentation for [setting up a development environ
 - All contributions should be documented, tested and linted. Please verify that all tests and lint checks pass successfully before proposing a change.
 - In order for a pull request to be accepted, the CI (containing unit tests, functional tests and linting) needs to succeed, in addition to approvals from two maintainers.
 - PRs should be small and solve/improve one issue at a time. If you have multiple suggestions for improvement, please open multiple PRs.
-
+- Every PR must bump the presidio version by changing the VERSION file to the appropriate [semver](https://semver.org/).
```
Comment (navalev) [inline CONTRIBUTING.md:30]: worth outlining guidlines to what would result in a change to major vs minor vs patch

**Review by omri374 — APPROVED** [review]

File: .pipelines/templates/validate-version.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        GIT_DIFF_UPSTREAMBRANCH=${{ parameters.UPSTREAM_BRANCH }}
+        if [[ ! $GIT_DIFF_UPSTREAMBRANCH == *"/"* ]]; then
+          echo "missing full path, adding remotes/origin/"
+          GIT_DIFF_UPSTREAMBRANCH=remotes/origin/$GIT_DIFF_UPSTREAMBRANCH
+        fi
+
+        GIT_DIFF_SOURCEBRANCH="HEAD"
+
+        VERSION_FILECHANGE_SET=$(git diff "$GIT_DIFF_SOURCEBRANCH" "$GIT_DIFF_UPSTREAMBRANCH" --name-only | grep -w VERSION)
+        # check if file was changeds
```
Comment (omri374) [inline .pipelines/templates/validate-version.yml:23]:
```suggestion
        # check if file has changed
```
nit: typo

File: .pipelines/templates/validate-version.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+steps:
+- task: Bash@3
+  displayName: 'verify version is changed'
+  name: verify
+  inputs:
+      targetType: 'inline'
+      script: |
+        set -eux  # exit on error
+    
+        GIT_DIFF_UPSTREAMBRANCH=${{ parameters.UPSTREAM_BRANCH }}
```
Comment (omri374) [inline .pipelines/templates/validate-version.yml:14]: Worth mentioning what this logic does? Comparing current branch with upstream branch? (master, right?)

File: .pipelines/templates/validate-version.yml (hunk 1 above)
Comment (SharonHart) [reply .pipelines/templates/validate-version.yml:26]: Then how will we change it major eventually if not by PR ?

Comment (balteravishay) [reply .pipelines/templates/validate-version.yml:26]: agree with @SharonHart . adding validation logic here will be very strict.
the PRs are also going through manual review, this is just to "not miss" a version bump by mistake.
I will open a user story for validating the version bump in CI (@shiranr also mentioned this in her PR).

**Review by navalev — DISMISSED** [review]

File: .pipelines/templates/validate-version.yml (hunk 1 above)
Comment (balteravishay) [reply .pipelines/templates/validate-version.yml:51]: added validation logic @SharonHart @shiranr @navalev :)

**Review by SharonHart — APPROVED** [review]

File: .pipelines/templates/validate-version.yml (hunk 1 above)
Comment (navalev) [reply .pipelines/templates/validate-version.yml:51]: looks good

**Review by shiranr — APPROVED** [review]

**Review by omri374 — APPROVED** [review]

**Pull request merged**